javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/resolution/SymbolReference.java

```diff
@@ -33,54 +34,81 @@
  */
 public class SymbolReference<S extends ResolvedDeclaration> {
 
-    private Optional<? extends S> correspondingDeclaration;
+    private final S correspondingDeclaration;
 
-    private SymbolReference(Optional<? extends S> correspondingDeclaration) {
+    private SymbolReference(@Nullable S correspondingDeclaration) {
         this.correspondingDeclaration = correspondingDeclaration;
     }
 
     /**
      * Create a solve reference to the given symbol.
      */
     public static <S extends ResolvedDeclaration, S2 extends S> SymbolReference<S> solved(S2 symbolDeclaration) {
-        return new SymbolReference<S>(Optional.of(symbolDeclaration));
+        return new SymbolReference<>(symbolDeclaration);
```

Comment:
Can this method declaration be simplified?

Reply:
It's a good question. Simplifying is not easy, since it impacts the api. For example, tried to do that in commit c4e42df but it may break the current API.